onnxruntime/core/providers/cpu/cpu_execution_provider.cc

```diff
@@ -751,6 +751,14 @@ class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain,
 class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 16, int32_t, LessOrEqual);
 class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 16, int64_t, LessOrEqual);
 
+// Opset 17
```

Comment:
No GPU?

Reply:
I don't know what the requirements are specifically for this kernel. My comment stems from the fact that if you have a model, that a customer wants to run on GPU and DFT is one of the few kernels (or the only one) that does not run on GPU, then it results in poor performance, because we need to copy data back to host, run CPU only kernel, then copy outputs back to GPU. The customer then files an issue. There was an effort to make sure that all kernels (as practical) run on GPU since it has proven to be such a frequent case.

Reply:
There is an effort spinning up now, where I will be implementing DirectML kernels for these ops. However, I do not know of an owner for cuda.

Reply:
> > static T compute_angular_velocity(size_t number_of_samples, bool inverse) {
> 
> Is this restricted to FP types?
> 
> Refers to: onnxruntime/core/providers/cpu/signal/dft.cc:149 in [62f6487](https://github.com/microsoft/onnxruntime/commit/62f64876ad25ebcff797512be3cfb8ec8b16353f). [](commit_id = [62f6487](https://github.com/microsoft/onnxruntime/commit/62f64876ad25ebcff797512be3cfb8ec8b16353f), deletion_comment = False)

compute_angular_velocity is templated in T to accept any numeric castable types.